sequencer/src/state.rs

```diff
@@ -210,112 +218,167 @@ impl ValidatedState {
     }
 }
 
+/// Possible proposal validation failures
+#[derive(Error, Debug, Eq, PartialEq)]
+pub enum ProposalValidationError {
+    #[error("Invalid ChainConfig: expected={expected}, proposal={proposal}")]
+    InvalidChainConfig { expected: String, proposal: String },
```

Comment:
How come we use strings here instead of just putting the `ChainConfig`?

Reply:
Using `String` is just an easy way to avoid[large enum varient](https://rust-lang.github.io/rust-clippy/master/index.html#/large_enum_variant) clippy warning. Since we are only using these error types in tests and for our own comprehension, this easy solution seems good enough to me.